### ISSUE TITLE
Dunfell support (Segmentation Fault)

### ISSUE THREAD

**Issue by michaelbaisch** [lucimber/meta-openjdk-temurin#21]
Hello,

when using this layer to deploy `17.0.9+9` on my dunfell installation to my "arm" board (raspberrypi) I'm getting a Segmentation Fault when executing `java -version`.
The behavior I'm seeing is exactly what is described in this stackoverflow thread: https://stackoverflow.com/questions/68025199/why-does-my-precompiled-binary-change-in-yocto
The binaries in `bin` get modified somehow, for me `java` grows from ~7 KB to ~10 KB. I only see the 7 KB version in the package work directory but in the rootfs directory there is the 10 KB version. I haven't been able to figure out what in yocto is changing it. Deploying it to `/usr/share` does the trick, but obviously now I get warnings that libraries are at the wrong location.

Greetings

**Comment by sebveit** [comment]
Thx for the feedback! I'll look into this next week. I'm using the arm64 "version" on rpi3 and I'm not seeing this segfault. Strange... :-/

**Comment by michaelbaisch** [comment]
I figured out the problem: it was the `image-prelink` task running during `do_image`. Prelink modifies the binaries in a list of directories, including `/usr/lib`, but not `/usr/share`. In newer versions of Yocto, prelink is not enabled by default, as it can sometimes cause issues and is not helpful for reproducible builds. So, I also disabled it by removing `image-prelink` from `USER_CLASSES` in my `local.conf`.
